compiler/dyno/lib/resolution/resolution-queries.cpp

```diff
@@ -1956,16 +1951,25 @@ doIsCandidateApplicableInitial(Context* context,
     return typeConstructorInitial(context, t);
   }
 
-  if (isVariable(tag) &&
-      ci.isParenless() && ci.isMethodCall() && ci.numActuals() == 1) {
-    // calling a field accessor
-    auto ct = ci.actual(0).type().type()->getCompositeType();
-    assert(ct);
-    assert(isNameOfField(context, ci.name(), ct));
-    return fieldAccessor(context, ct, ci.name());
+  if (isFormal(tag)) {
```

Comment:
This isn't an issue with the PR, but maybe with my understanding. Supposing that we have first-class functions (via a lambda or something), and we passed such a function as a formal, and this function is more specific than the other. Wouldn't this prevent it from matching as a candidate?

```Chapel

var someFunction = lambda(x: int): void {
    writeln("More specific");
};

class C {
    proc test(x): void {
        writeln("Less specific");
    }

    proc runTest(test: someFunction.type) {
        test(1);
    }
}

var c = new C();
c.runTest(someFunction); // Prints "More specific"
```

This isn't a method call like `x.T` as the PR title describes, but I don't see any way that this `isCandidateApplicableInitial` code differentiates from that case. Maybe a `isMethodCall()` needs to be specified too?

Reply:
unless, of course, the candidate would be the `Identifier` itself, rather than the `test : ...` formal.

Reply:
I'm not sure, I think we should worry about changing the code to handle function types when we get to it.

Reply:
Well, the above works right now in the production compiler, so it's not like I'm inventing a hypothetical future failure mode.

Reply:
I would guess that the reason it prints `More specific` is because the language rules prefer locals to fields in the presence of multiple candidates. So it's a consequence of name resolution rather than typing. The current implementation of lambdas can't support passing a generic lambda (we haven't decided what the means yet) at the moment.